include/astarte_device.h

```diff
@@ -56,6 +56,7 @@ typedef void (*astarte_device_disconnection_event_callback_t)(
 typedef struct
 {
     astarte_device_data_event_callback_t data_event_callback;
+    astarte_device_data_event_callback_t unset_data_event_callback;
```

Comment:
let's define a `astarte_device_unset_event_t`, so we don't expose a bogus pointer.